### ISSUE TITLE
Improve batch wrapper

### ISSUE THREAD

**Issue by ErikEJ** [ErikEJ/SqlServer.Rules#753]
This ad-hoc script currently fails completely:

```sql
CREATE TABLE dbo.test (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

DECLARE @someTable TABLE (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

UPDATE dst
SET currentId = bodyItems.currentId
FROM dbo.test dst
INNER JOIN @someTable bodyItems
ON bodyItems.originalId = dst.originalId;
```
The reason is that this script is a single mixed batch with CREATE TABLE + ad-hoc DML and no GO. BatchWrapper only wraps batches when all statements are wrappable; once CREATE TABLE is present, the batch is left untouched and then DacFx rejects it with SQL71006 (Only one statement is allowed per batch). Because that batch never gets wrapped into the synthetic procedure, neither SRD0018 nor SRP0014 runs on it.

The obvious improvement would be to make BatchWrapper handle mixed batches incrementally instead of all-or-nothing per parsed batch: leave CREATE TABLE unwrapped, then wrap the following wrappable ad-hoc statements separately. That would let the later DECLARE/UPDATE still be analyzed, instead of the whole batch falling through to SQL71006 and producing no diagnostics.

**Comment by ErikEJ** [comment]
@niphlod FYI

**Comment by ErikEJ** [comment]
@niphlod Confirmed that this bug is fixed now, thanks for reporting
